auth-api/src/test/java/io/stargate/auth/api/AuthResourceTest.java

```diff
@@ -0,0 +1,244 @@
+    }
+
+    @Test
+    void createTokenFromCredentialsEmptyUsername() {
+        Credentials credentials = new Credentials("", "password");
+
+        Response response = resourceWithUsernameTokenDisabled.target("/v1/auth")
+                .request()
+                .post(Entity.entity(credentials, MediaType.APPLICATION_JSON));
+
+        assertThat(response.getStatusInfo().getStatusCode()).isEqualTo(400);
+    }
+
+    @Test
+    void createTokenFromCredentialsEmptyPassword() {
+        Credentials credentials = new Credentials("username", "");
+
+        Response response = resourceWithUsernameTokenDisabled.target("/v1/auth")
+                .request()
+                .post(Entity.entity(credentials, MediaType.APPLICATION_JSON));
+
+        assertThat(response.getStatusInfo().getStatusCode()).isEqualTo(400);
+    }
+
+    @Test
+    void createTokenFromUsernameSuccess() throws UnauthorizedException {
+        UsernameCredentials username = new UsernameCredentials("username");
+        when(authService.createToken("username")).thenReturn("token");
+
+        AuthTokenResponse token = resourceWithUsernameTokenEnabled.target("/v1/admin/auth/usernametoken")
+                .request()
+                .post(Entity.entity(username, MediaType.APPLICATION_JSON), AuthTokenResponse.class);
+
+        assertThat(token.getAuthToken()).isEqualTo("token");
+    }
+
+    @Test
+    void createTokenFromUsernameDisabled() throws UnauthorizedException {
+        UsernameCredentials username = new UsernameCredentials("username");
+        when(authService.createToken("username")).thenReturn("token");
+
+        Response response = resourceWithUsernameTokenDisabled.target("/v1/admin/auth/usernametoken")
+                .request()
+                .post(Entity.entity(username, MediaType.APPLICATION_JSON));
+
+        assertThat(response.getStatusInfo().getStatusCode()).isEqualTo(400);
+    }
+
+    @Test
+    void createTokenFromUsernameUnauthorized() throws UnauthorizedException {
+        UsernameCredentials username = new UsernameCredentials("username");
+        when(authService.createToken("username")).thenThrow(UnauthorizedException.class);
+
+        Response response = resourceWithUsernameTokenEnabled.target("/v1/admin/auth/usernametoken")
+                .request()
+                .post(Entity.entity(username, MediaType.APPLICATION_JSON));
+
+        assertThat(response.getStatusInfo().getStatusCode()).isEqualTo(401);
+    }
+
+    @Test
+    void createTokenFromUsernameInternalServerError() throws UnauthorizedException {
+        UsernameCredentials username = new UsernameCredentials("username");
+        when(authService.createToken("username")).thenThrow(RuntimeException.class);
+
+        Response response = resourceWithUsernameTokenEnabled.target("/v1/admin/auth/usernametoken")
+                .request()
+                .post(Entity.entity(username, MediaType.APPLICATION_JSON));
```

Comment:
These three lines are repeated many times with minor variations... what do you think about introducing a utility method for this code (with params) to make the test methods less verbose?

Reply:
I thought about it. But I preferred readable test code over introducing additional abstractions. The resource is sometimes different, the target URL, the entity. So the only thing that is 100% reusable is `.request()` and the media type and the fact that it's a POST request.

I'd leave it like this for now if it's ok?

Reply:
How about something like this new method local to this class:
```
Response post(String url, Object entity, boolean withToken) { ... }
```
The calling code in this case would be:
```
Response response = post(""/v1/admin/auth/usernametoken", username, true); // one line instead of three
```
WDYT?

Reply:
Sure.

Reply:
thx! :)

Reply:
Ah, well, not much savings in terms of lines :( Feel free to use the style you prefer in this case :)